src/TokenFuzzChecks.sol

```diff
@@ -0,0 +1,461 @@
+        fuzzCheckTokenModifiers(_token, _contractName, who);
+    }
+
+
+    function fuzzCheckMint(
+        address _token,
+        string memory _contractName,
+        address to,
+        uint256 mintAmount
+    ) public {
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        uint256 prevToBalance = TokenLike(_token).balanceOf(to);
+        mintAmount = bound(mintAmount, 0, type(uint256).max - prevSupply);
+        uint256 prevWard = TokenLike(_token).wards(address(this));
+        _token.setWard(address(this), 1);
+        if (to != address(0) && to != _token) {
+            vm.expectEmit(true, true, true, true);
+            emit Transfer(address(0), to, mintAmount);
+        } else {
+            vm.expectRevert(abi.encodePacked(_contractName, "/invalid-address"));
+        }
+
+        TokenLike(_token).mint(to, mintAmount);
+
+        if (to != address(0) && to != _token) {
+            assertEq(TokenLike(_token).totalSupply(), prevSupply + mintAmount);
+            assertEq(TokenLike(_token).balanceOf(to), prevToBalance + mintAmount);
+        }
+        _token.setWard(address(this), prevWard);
+    }
+
+    function fuzzCheckBurn(
+        address _token,
+        address from,
+        uint256 mintAmount,
+        uint256 burnAmount
+    ) public {
+        vm.assume(from != address(0) && from != _token);
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        uint256 prevFromBalance = TokenLike(_token).balanceOf(from);
+        mintAmount = bound(mintAmount, 0, type(uint256).max - prevSupply);
+        burnAmount = bound(burnAmount, 0, mintAmount);
+        deal(_token, from, TokenLike(_token).balanceOf(from) + mintAmount, true);
+
+        vm.expectEmit(true, true, true, true);
+        emit Transfer(from, address(0), burnAmount);
+        vm.prank(from);
+        TokenLike(_token).burn(from, burnAmount);
+
+        assertEq(TokenLike(_token).totalSupply(), prevSupply + mintAmount - burnAmount);
+        assertEq(TokenLike(_token).balanceOf(from), prevFromBalance + mintAmount - burnAmount);
+    }
+
+    function fuzzCheckBurnInsufficientBalance(
+        address _token,
+        string memory _contractName,
+        address to,
+        uint256 mintAmount,
+        uint256 burnAmount
+    ) public {
+        vm.assume(to != address(0) && to != _token);
+        uint256 prevSupply = TokenLike(_token).totalSupply();
+        mintAmount = bound(mintAmount, 0, type(uint256).max - prevSupply - 1);
+        burnAmount = bound(burnAmount, mintAmount + 1, type(uint256).max);
+        deal(_token, to, TokenLike(_token).balanceOf(to) + mintAmount, true);
+
+        vm.expectRevert(abi.encodePacked(_contractName, "/insufficient-balance"));
+        TokenLike(_token).burn(to, burnAmount);
+    }
+
+    function fuzzCheckTokenModifiers(
+        address _token,
+        string memory _contractName,
+        address sender
+    ) public {
+        vm.assume(sender != address(this));
```

Comment:
Is this actually correct? As it is assuming the address which will be permissioned is `address(this)` which shouldn't be the case anymore.
Is it possible to do something like this: `vm.assume(TokenLike(_token).wards(sender) == 0)`? (have never used this forge hack).

Reply:
`address(this)` _could_ still be permissionned in the lib user's test suite but yeah I think what you propose is better. Changed.

Reply:
Right, my point was more about if for a specific implementation the permissioned address was another than `address(this)`, then in that case we could have a failure. Anyway it is changed now and looks good.